src/pre_compute/signer.rs

```diff
@@ -0,0 +1,179 @@
+/// let private_key = "0xdd3b993ec21c71c1f6d63a5240850e0d4d8dd83ff70d29e49247958548c1d479";
+///
+/// match sign_enclave_challenge(message_hash, private_key) {
+///     Ok(signature) => println!("Signature: {}", signature),
+///     Err(e) => eprintln!("Error: {:?}", e),
+/// }
+/// ```
+pub fn sign_enclave_challenge(
+    message_hash: &str,
+    enclave_challenge_private_key: &str,
+) -> Result<String, PreComputeError> {
+    let signer: PrivateKeySigner = enclave_challenge_private_key
+        .parse::<PrivateKeySigner>()
+        .map_err(|_| {
+            PreComputeError::new(ReplicateStatusCause::PreComputeInvalidEnclaveChallengePrivateKey)
+        })?;
+
+    let signature: Signature = signer
+        .sign_message_sync(&hex_string_to_byte_array(&message_hash))
+        .map_err(|_| PreComputeError::new(ReplicateStatusCause::PreComputeInvalidTeeSignature))?;
+
+    Ok(signature.to_string())
+}
+
+/// Generates a challenge signature for a given chain task ID.
+///
+/// This function retrieves the worker address and TEE challenge private key from the environment,
+/// then creates a message hash by concatenating and hashing the chain task ID and worker address.
+/// Finally, it signs this message hash with the private key.
+///
+/// # Arguments
+///
+/// * `chain_task_id` - A string identifier for the chain task
+///
+/// # Returns
+///
+/// * `Ok(String)` - The challenge signature as a hexadecimal string if successful
+/// * `Err(PreComputeError)` - An error if required environment variables are missing or if signing fails
+///
+/// # Errors
+///
+/// This function will return an error in the following situations:
+/// * The worker address environment variable is missing (returns `PreComputeWorkerAddressMissing`)
+/// * The TEE challenge private key environment variable is missing (returns `PreComputeTeeChallengePrivateKeyMissing`)
+/// * The signing operation fails (returns `PreComputeInvalidTeeSignature`)
+///
+/// # Environment Variables
+///
+/// * `SIGN_WORKER_ADDRESS` - The worker's address used in message hash calculation
+/// * `SIGN_TEE_CHALLENGE_PRIVATE_KEY` - The private key used for signing the challenge
+///
+/// # Example
+///
+/// ```
+/// // Assuming the necessary environment variables are set:
+/// // SIGN_WORKER_ADDRESS=0xabcdef123456789
+/// // SIGN_TEE_CHALLENGE_PRIVATE_KEY=0xdd3b993ec21c71c1f6d63a5240850e0d4d8dd83ff70d29e49247958548c1d479
+///
+/// let chain_task_id = "0x123456789abcdef";
+///
+/// match challenge(chain_task_id) {
+///     Ok(signature) => println!("Challenge signature: {}", signature),
+///     Err(e) => eprintln!("Error generating challenge: {:?}", e),
+/// }
+/// ```
+pub fn get_challenge(chain_task_id: &str) -> Result<String, PreComputeError> {
+    let worker_address = get_env_var_or_error(
+        TeeSessionEnvironmentVariable::SIGN_WORKER_ADDRESS,
+        ReplicateStatusCause::PreComputeInvalidEnclaveChallengePrivateKey,
```

Comment:
Is the `PRE_COMPUTE_WORKER_ADDRESS_MISSING` entry mussing ?
```suggestion
        ReplicateStatusCause::PreComputeWorkerAddressMissing,
```